### ISSUE TITLE
Breaks when used with Python2

### ISSUE THREAD

**Issue by jelmer** [Kozea/Radicale#282]
Radicale currently seems to break with Python2:

Traceback (most recent call last):
  File "/usr/lib/python2.7/wsgiref/handlers.py", line 85, in run
    self.result = application(self.environ, self.start_response)
  File "/home/jelmer/Radicale/radicale/__init__.py", line 312, in __call__
    self.collect_allowed_items(items, user)
  File "/home/jelmer/Radicale/radicale/__init__.py", line 195, in collect_allowed_items
    if rights.authorized(user, item, "r"):
  File "/home/jelmer/Radicale/radicale/rights/regex.py", line 111, in authorized
    _read_from_sections(user or "", collection_url, permission))
  File "/home/jelmer/Radicale/radicale/rights/regex.py", line 72, in _read_from_sections
    regex.readfp(StringIO(DEFINED_RIGHTS[rights_type]))
TypeError: initial_value must be unicode or None, not str

Reproduced with both master and 0.9.

**Comment by liZe** [comment]
I can't reproduce. Can you try this code on your version of Python?

> from StringIO import StringIO
> StringIO(b'test')

And this code:

> from configparser import ConfigParser

The second code sample should't work on Python 2, as the module is called ConfigParser and not configparser.

But let me guess: you're on MacOS or Windows, the imports are case-insensitive, the second code sample works and `StringIO` is imported from `io` instead of `StringIO`, and `io.StringIO` needs an unicode object (`StringIO.StringIO` works with bytes).

TL;DR: LOL case-insensitive LOL python LOL MacOS/Windows LOL

**Comment by jelmer** [comment]
On Wed, Apr 29, 2015 at 09:53:38AM -0700, Guillaume Ayoub wrote:
> I can't reproduce. Can you try this code on your version of Python?
> 
> > from StringIO import StringIO
> > StringIO(b'test')

Python 2.7.9 (default, Mar  1 2015, 12:57:24) 
[GCC 4.9.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from StringIO import StringIO
>>> StringIO(b'test')
<StringIO.StringIO instance at 0x7fcd839ded40>

The problem appears to be this code on top of Radicale/rights/regex.py:

try:
    from configparser import ConfigParser
    from io import StringIO
except ImportError:
    from ConfigParser import ConfigParser
    from StringIO import StringIO

> But let me guess: you're on MacOS or Windows, the imports are case-insensitive, the second code sample works and `StringIO` is imported from `io` instead of `StringIO`, and `io.StringIO` needs an unicode object (`StringIO.StringIO` works with bytes).
Nope :) I'm on Debian sid. However, I have a python module with the name "configparser" installed. This is the module with the same name from Python3 backported to Python2 for convenience - https://pypi.python.org/pypi/configparser.

Cheers,

Jelmer

**Comment by jelmer** [comment]
On Wed, Apr 29, 2015 at 10:05:53AM -0700, Guillaume Ayoub wrote:
> Closed #282 via e69bec127295ff9bd323140bf1ed7d62da580914.
Thanks for the quick fix!
